src/NetAnalyzers/Core/Microsoft.CodeQuality.Analyzers/QualityGuidelines/AssigningSymbolAndItsMemberInSameStatement.Fixer.cs

```diff
@@ -0,0 +1,122 @@
+using Microsoft.CodeAnalysis.CodeActions;
+using Microsoft.CodeAnalysis.CodeFixes;
+using Microsoft.CodeAnalysis.CSharp;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+
+#nullable enable
+
+namespace Microsoft.CodeQuality.Analyzers.QualityGuidelines
+{
+    [ExportCodeFixProvider(LanguageNames.CSharp, Name = AssigningSymbolAndItsMemberInSameStatement.RuleId), Shared]
+    public sealed class AssigningSymbolAndItsMemberInSameStatementFixer : CodeFixProvider
+    {
+        public override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(AssigningSymbolAndItsMemberInSameStatement.RuleId);
+
+        public override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            // For something like `a.x = a = b;`, we offer 3 code fixes:
+            // First:
+            //     a = b;
+            //     a.x = b;
+            // Second:
+            //     a = b;
+            //     a.x = a;
+            // Third: (Not currently implemented)
+            //     var temp = a;
+            //     a = b;
+            //     temp.x = b;
+
+            var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+            if (root.FindNode(context.Span).Parent is not AssignmentExpressionSyntax assignment)
+            {
+                return;
+            }
+
+            var members = GetAssignmentMembers(assignment);
+            if (members.Count < 3)
+            {
+                return;
+            }
+
+            var mostRightMember = members.Peek(); // Don't move this near its usage. The result will be different.
+            var leadingTrivia = assignment.Parent.GetLeadingTrivia();
+            var trailingTrivia = assignment.Parent.GetTrailingTrivia();
+            var replacements = new List<SyntaxNode>(members.Count - 1);
+
+            while (members.Count > 2)
+            {
+                replacements.Add(GetAssignmentExpressionStatement(members, leadingTrivia, trailingTrivia));
+                trailingTrivia = SyntaxTriviaList.Empty; // Take the trailing trivia on the first assignment only.
+            }
+
+            var title = MicrosoftCodeQualityAnalyzersResources.AssigningSymbolAndItsMemberInSameStatementTitle;
+
+            var replacements1 = replacements.Concat(GetAssignmentExpressionStatement(assignment.Left, mostRightMember, leadingTrivia, trailingTrivia));
+            var replacements2 = replacements.Concat(GetAssignmentExpressionStatement(members, leadingTrivia, trailingTrivia));
+
+#pragma warning disable RS1010 // Provide an explicit value for EquivalenceKey - false positve 
+            var nestedCodeAction = CodeAction.Create(title, ImmutableArray.Create<CodeAction>(
+                new MyCodeAction($"{title} 1", ct => GetDocument(context.Document, root, assignment.Parent, replacements1)),
+                new MyCodeAction($"{title} 2", ct => GetDocument(context.Document, root, assignment.Parent, replacements2))
```

Comment:
The `title` parameter is what get displayed so it feels off to have `XXX 1` and `XXX 2`.

Reply:
@Evangelink I have this temporary until I can get with good titles. Do you have any suggestions? Not sure how to describe the code fixes.